### PR TITLE
feat: Add updated UI for Snap install and update flows

### LINKED ISSUE #23244: Facelifting installation & update screens
Let's match our snaps lifecycle screens with the confirmation redesign (and our upcoming dynamic permission request screen)

facelifted-snaps-flows

### PR DESCRIPTION
**Pull request by david0xd**
## **Description**
This PR updates UI for Snap Install and Snap update flow. Alongside the update, replaced are deprecated Design System components.

Updates are made respecting the designs presented on Figma: https://www.figma.com/file/jr7XsFvcPCnf4HOMvNWFfn/Permission-System?node-id=1%3A2&mode=dev

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23870?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/metamask-extension/issues/23244

## **Manual testing steps**
1. Start Snap installation process and check the UI.
2. Start Snap update process and check the UI.

### **Before**
Screenshot 2024-04-05 at 22 36 20
Screenshot 2024-04-05 at 22 36 33
Screenshot 2024-04-05 at 22 36 56
Screenshot 2024-04-05 at 22 37 18

### **After**
Screenshot 2024-04-05 at 21 37 23
Screenshot 2024-04-05 at 21 37 36
Screenshot 2024-04-05 at 21 37 46
Screenshot 2024-04-05 at 21 38 25

## **Pre-merge author checklist**
- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**
- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by bowensanders** [comment]
OK, I've pushed fixes for all the E2E tests.
Side Quest: Figure out why the accounts/create-snap-account test was flaking. Seem to have fixed that here as well.

**Review by bowensanders — DISMISSED** [review]
LGTM ✅

File: ui/pages/permissions-connect/snaps/snap-install/snap-install.js
Comment (FrederikBolding) [inline ui/pages/permissions-connect/snaps/snap-install/snap-install.js:1]: Should we add a little space at the bottom so you can scroll past then end?
image

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js [hunk 1]
```diff
@@ -168,9 +175,7 @@ export default function SnapResult({
         className="snap-result__footer"
         alignItems={AlignItems.center}
         flexDirection={FlexDirection.Column}
-        style={{
```
Comment (FrederikBolding) [inline ui/pages/permissions-connect/snaps/snap-result/snap-result.js:171]: Is it intentional to drop the box shadow for the "scroll area overlay"? 🤔

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ export default function SnapPermissionsList({
   const snapsMetadata = useSelector(getSnapsMetadata);
 
   return (
-    <Box paddingTop={2} paddingBottom={2} className="snap-permissions-list">
+    <Box paddingTop={1} paddingBottom={1} className="snap-permissions-list">
```
Comment (FrederikBolding) [inline ui/components/app/snaps/snap-permissions-list/snap-permissions-list.js:21]: Looks like this has no padding in the design but has extra padding in the implementation
image

File: ui/pages/permissions-connect/snaps/snap-result/snap-result.js (hunk 1 above)
Comment (eriknson) [reply ui/pages/permissions-connect/snaps/snap-result/snap-result.js:171]: Yes, we're dropping the visual container for the footer

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**